### PR TITLE
Run all tests even if some fail

### PR DESCRIPTION
**Pull request by peastman**
Hopefully this fixes the problem in Github Actions where if one test stage fails, it doesn't run any of the later ones.  @jaimergp can you check that the mystical incantations look correct?

### CODE REVIEW

**Comment by jaimergp** [comment]
Hm, this is not working as I thought.

There are two ways: using `always()` like specified [here](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions) or using [`job.steps.step_id.outcome`](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#steps-context). Problem with `always()` is that the job will be marked as successful even if a step failed (I think?), so that's why use `outcome`s. But for this you also need to set [`continue-on-error`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) in the previous steps.

Or maybe it's the other way around, but that's the two concepts we need to play with.

**Comment by peastman** [comment]
This page looks relevant?

https://github.community/t/workflow-skips-steps-besides-if-condition/115956

It indicates that when we write

```
if: ${{ steps.yapf.outcome == 'failure' || steps.yapf.outcome == 'success' }}
```

it instead gets interpreted as

```
if: ${{ success() && (steps.yapf.outcome == 'failure' || steps.yapf.outcome == 'success') }}
```

It isn't clear to me what the correct solution is.  Perhaps include `(success() || failure())` as part of the condition?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -129,18 +130,27 @@ jobs:
           yapf -d $CHANGED_FILES
         fi
     - name: Flake8
+      id: flake8
```
Comment (jaimergp) [inline .github/workflows/main.yml:133]: Maybe with something like this?

```suggestion
      id: flake8
      continue-on-error: true
```

Comment (peastman) [reply .github/workflows/main.yml:133]: I saw that option in the documentation, but the description is unclear.  If we use it, will that cause it to report that the run succeeded even if that step failed?

**Comment by peastman** [comment]
As I suspected, `continue-on-error` doesn't work.  That causes it to report success even if tests failed.

**Comment by peastman** [comment]
Looks like it's working!  `test_mol2vec_fingerprint()` is failing (as it is on the main branch), but it still continues on to upload the coverage report.

**Comment by jaimergp** [comment]
> Make tests depend on install rather than previous tests

Wow this is such a nice way to reason about the pipeline! I was somehow forcing myself to think linearly, that this makes so much more sense! Glad to see you got it working!

**Comment by rbharath** [comment]
This is great! Feel free to merge in once ready :)

**Pull request merged**